### PR TITLE
tweak jumping

### PR DESCRIPTION
**Pull request by townofdon**
- add variable jump height based on jump button hold
- add terminal velocity
- fix bouncy mushroom inconsistent super jump

### CODE REVIEW

File: Assets/ScriptableObjects/MovementOverride.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ public class MovementOverride : ScriptableObject
     [Range(0, 5)] public int maxAirJumps = 0;
     public float downwardMovementMultiplier = 10f;
     public float upwardMovementMultiplier = 7f;
+    public float upwardMovementShortJumpMultiplier = 15f;
+    public float terminalVelocity = 3f; // multiple of gravity constant, 9.8 m/s
```
Comment (MitchellCoon) [inline Assets/ScriptableObjects/MovementOverride.cs:22]: It's 9.8 m/s^2, jeez

File: Assets/Scripts/Capabilities/Jump.cs [hunk 2]
```diff
@@ -72,26 +72,37 @@ private void FixedUpdate()
...
         }
 
         body.velocity = velocity;
 
-
+        // clamp fall speed to terminal velocity
+        if (body.velocity.y < 0)
+        {
+            float clampedYSpeed = Mathf.Clamp(body.velocity.y, -Constants.GRAVITY * movement.terminalVelocity, 0);
+            body.velocity = new Vector2(body.velocity.x, clampedYSpeed);
```
Comment (MitchellCoon) [inline Assets/Scripts/Capabilities/Jump.cs:104]: Minor thing, can we set body.velocity outside of the if block and then just remove the earlier line so we're not setting it twice?

File: Assets/Scripts/Capabilities/Jump.cs [hunk 3]
```diff
@@ -117,13 +128,10 @@ private void JumpAction()
             }
             if (!isGrounded && velocity.y < 0)
             {
-                velocity.y = jumpSpeed;
-            }
-            else
-            {
-                velocity.y += jumpSpeed;
+                // first zero out y velocity
```
Comment (MitchellCoon) [inline Assets/Scripts/Capabilities/Jump.cs:131]: If I had to guess, that's probably what fixed the super jump. Thanks for ruining it >.>

Comment (MitchellCoon) [reply Assets/Scripts/Capabilities/Jump.cs:131]: JK, it's still happening

File: Assets/ScriptableObjects/MovementOverride.cs (hunk 1 above)
Comment (townofdon) [reply Assets/ScriptableObjects/MovementOverride.cs:22]: eh, semantics - accel vs. velocity 😁

File: Assets/Scripts/Capabilities/Jump.cs (hunk 2 above)
Comment (townofdon) [reply Assets/Scripts/Capabilities/Jump.cs:104]: We def can - is there a larger reason than code organization for this? e.g. if calling `body.velocity.set` does some extra internal stuff that would make this less performant?

Comment (MitchellCoon) [reply Assets/Scripts/Capabilities/Jump.cs]: Just organization

**Pull request merged**